src/chunkserver/datastore/file_pool.cpp

```diff
@@ -375,6 +401,128 @@ void FilePool::CleanWorker() {
     }
 }
 
+bool FilePool::PrepareFormat() {
+    curve::fs::FileSystemInfo finfo;
+    int r = fsptr_->Statfs(currentdir_, &finfo);
+    if (r != 0) {
+        LOG(ERROR) << "get disk usage info failed!";
+        return false;
+    }
+
+    if (poolOpt_.allocatedByPercent) {
+        poolOpt_.filePoolSize = finfo.total * poolOpt_.allocatedPercent / 100;
+    }
+
+    uint64_t bytesPerPage = poolOpt_.fileSize + poolOpt_.metaFileSize;
+    uint64_t needSpace = poolOpt_.filePoolSize -
+                         currentState_.chunkNum * bytesPerPage;
+    uint64_t vaildSpace = 0;
+
+    if (poolOpt_.filePoolSize / bytesPerPage < currentState_.chunkNum) {
+        LOG(INFO) << "It is no need to format chunks.";
+        formatStat_.preAllocateNum = 0;
+        formatStat_.allocateChunkNum = 0;
+        return true;
+    }
+
+    vaildSpace = finfo.available;
+    LOG(INFO) << "free space = " << finfo.available
+            << ", total space = " << finfo.total
+            << ", need space = " << needSpace;
+    if (vaildSpace < needSpace) {
+        LOG(ERROR) << "disk free space not enough.";
+        return false;
+    }
+
+    formatStat_.preAllocateNum = needSpace / bytesPerPage;
+    formatStat_.allocateChunkNum = 0;
+    LOG(INFO) << "preAllocateNum = " << formatStat_.preAllocateNum;
+    return true;
+}
+
+bool FilePool::WaitFormatDone() {
+    std::unique_lock<std::mutex> lk(mtx_);
```

Comment:
if this function is just for testing, it's better to add a suffix `ForTesting`, like `WaitFormatDoneForTesting`